qt/widgets/common/src/WorkspacePresenter/WorkspaceTreeWidgetSimple.cpp

```diff
@@ -185,7 +192,41 @@ void WorkspaceTreeWidgetSimple::popupContextMenu() {
     } else if (std::dynamic_pointer_cast<IMDWorkspace>(workspace)) {
       menu->addAction(m_showAlgorithmHistory);
       menu->addAction(m_sampleLogs);
-      menu->addAction(m_sliceViewer);
+
+      // launch slice viewer or plot spectrum conditionally
+      bool add_slice_viewer = true;
+
+      if (std::dynamic_pointer_cast<IMDHistoWorkspace>(workspace)) {
+        auto mdhist_ws =
+            std::dynamic_pointer_cast<IMDHistoWorkspace>(workspace);
+        // if the number of non-integral  if the number of non-integrated
+        // dimensions is 1.
+        auto num_dims = mdhist_ws->getNumDims();
+        if (num_dims == 1) {
+          // only 1 dimension
+          add_slice_viewer = false;
+        } else {
+          // multiple dimension: check number of non-integral
+          auto num_no_integral_dim = 0;
+          for (size_t d = 0; d < num_dims; ++d) {
+            if (mdhist_ws->getDimension(d)->getNBins() > 1)
+              ++num_no_integral_dim;
+          }
+          // if non-integral dimension is only 1, then enable to plot spectrum
+          if (num_no_integral_dim == 1)
+            add_slice_viewer = false;
+          else if (num_no_integral_dim == 0)
+            throw std::runtime_error(
```

Comment:
This will crash workbench :( Try: `CreateMDHistoWorkspace(SignalInput='1', ErrorInput='1', Dimensionality=2, Extents='-1,1,-1,1', NumberOfBins='1,1', Names='a,b', Units='a,b', OutputWorkspace='w')`. Right clicking willcause a crash. For now, the simplest solution is to not do anything. SliceViewer will nicely complain about non-integrated dimensions, but that's OK. @martyngigg Do you have an idea why this is not handled correctly?